File: XYMenuDemo/XYMenu/RootViewController.m
//
//  RootViewController.m
//  XYMenu
//
//  Created by FireHsia on 2018/2/2.
//  Copyright © 2018年 FireHsia. All rights reserved.
//

#import "RootViewController.h"
#import "UIBarButtonItem+XYMenu.h"

@interface RootViewController ()

@end

@implementation RootViewController

- (void)viewDidLoad {
    [super viewDidLoad];

}
- (void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
    [XYMenu dismissMenuInView:self.view];;
}

- (IBAction)leftBar:(id)sender {
    UIBarButtonItem *item = (UIBarButtonItem *)sender;
    NSArray *imageArr = @[@"swap", @"selected", @"code"];
    NSArray *titleArr = @[@"扫一扫", @"拍    照", @"付款码"];
    [item xy_showMenuWithImages:imageArr titles:titleArr menuType:XYMenuLeftNavBar currentNavVC:self.navigationController withItemClickIndex:^(NSInteger index) {
        [self showMessage:index];
    }];
    
}
- (IBAction)rightBar:(id)sender {
    UIBarButtonItem *item = (UIBarButtonItem *)sender;
    NSArray *imageArr = @[@"swap", @"selected", @"code"];
    NSArray *titleArr = @[@"扫一扫", @"拍    照", @"付款码"];
    [item xy_showMenuWithImages:imageArr titles:titleArr menuType:XYMenuRightNavBar currentNavVC:self.navigationController withItemClickIndex:^(NSInteger index) {
        [self showMessage:index];
    }];
}



- (void)showAlertMessage:(NSString *)message
{
    UIAlertController *alertVC = [UIAlertController alertControllerWithTitle:@"" message:message preferredStyle:UIAlertControllerStyleAlert];
    [alertVC addAction:[UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        
    }]];
    [self presentViewController:alertVC animated:YES completion:nil];
}

- (void)showMessage:(NSInteger)index
{
    switch (index) {
        case 1:
        {
            [self showAlertMessage:@"扫一扫"];
        }
            break;
        case 2:
        {
            [self showAlertMessage:@"拍   照"];
        }
            break;
        case 3:
        {
            [self showAlertMessage:@"付款码"];
        }
            break;
        default:
            break;
    }
}






@end

File: XYMenuSwiftDemo/XYMenuSwiftDemo/RootViewController.swift
//
//  RootViewController.swift
//  XYMenuSwiftDemo
//
//  Created by FireHsia on 2018/2/2.
//  Copyright © 2018年 FireHsia. All rights reserved.
//

import UIKit

class RootViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

    }

    @IBAction func leftBarItem(_ sender: Any) {
        self.showXYMenu(sender: sender, type: .XYMenuLeftNavBar, isNav: true)
    }
    
    @IBAction func rigthBarItem(_ sender: Any) {
        self.showXYMenu(sender: sender, type: .XYMenuRightNavBar, isNav: true)
    }
}


extension UIViewController {
    
    func showMessage(index: Int, titles: [String]) {
        let title = titles[index - 1]
        let alert = UIAlertController(title: title, message: "", preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "确定", style: .default, handler: { (action) in
            alert.dismiss(animated: true, completion: nil)
        }))
        self.present(alert, animated: true, completion: nil)
    }
    
    func showXYMenu(sender: Any, type: XYMenuType, isNav: Bool) {
        let images = ["code", "selected", "swap"]
        let titles = ["付款码", "拍    照", "扫一扫"]
        if isNav {
            if let barButtonItem = sender as? UIBarButtonItem {
                barButtonItem.xy_showXYMenu(images: images, titles: titles, currentNavVC: self.navigationController!, type: type, closure: { [unowned self] (index) in
                    self.showMessage(index: index, titles: titles)
                })
            }
        }else {
            if let senderView = sender as? UIView {
                senderView.xy_showXYMenu(images: images, titles: titles, type: type, closure: { [unowned self] (index) in
                    self.showMessage(index: index, titles: titles)
                })
            }
        }
    }
}
